In [26]:
import numpy as np
import pandas as pd
import plotly.express as px
from collections import Counter

Import data

In [6]:
train_df_og = pd.read_csv("./data/train.csv")
test_df_og = pd.read_csv("./data/test.csv")

Lets explore the data

In [17]:
train_df.head(5)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [34]:
train_df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

Key insights:
1. The dataset is quite small having only 891 entries
1. Baseline for the model: Predict that none passangers survived and it will yield accuracy of 42%, since we see that only 38% of passengers survived
1. Not all entries contain ages of passengers
1. On average 1 in 2 passengers had a sibling or spouse on board and 1 in 3 have a family member (blood related)

In [24]:
train_df.isna().sum().to_frame().T

PassengerId  Survived  Pclass  Name  Sex  Age  SibSp  Parch  Ticket  Fare  \
0            0         0       0     0    0  177      0      0       0     0   

   Cabin  Embarked  
0    687         2

Missing alot of cabain numbers, but it such information is most likeley not useful for prediction anyways, as we have better variables that suggest about the cabains location or quality - Fare and Ticket class.

Age on the other hand might be a crucial feature, based on the rule "Childern and Women first". We should try some solutions for it. Removing missing data does not seem good as it accounts for 20% of training data. I will try to populate with average value and create a predictor

As for Embarked, only 2 missing data points, we can easily remove or assign the mode value, as i fail to see that this could be a crucial feature for predictions

In [32]:
Counter(train_df['Embarked'])

Counter({'S': 644, 'C': 168, 'Q': 77, nan: 2})

In [29]:
Counter(train_df['Sex'])

Counter({'male': 577, 'female': 314})